src/bam/record.rs

```diff
@@ -321,22 +320,29 @@ impl Record {
         let q_len = qname.len() + 1;
         let extranul = extranul_from_qname(qname);
 
-        self.inner_mut().l_data = (q_len
-            + extranul
-            + cigar_width
-            + ((seq.len() as f32 / 2.0).ceil() as usize)
-            + qual.len()) as i32;
-
-        assert!(qname.len() <= 256);
```

Comment:
we do probably still want this assert _somewhere_.

Reply:
I've moved it up to new line 310. `assert!(qname.len() < 255);` The SAM spec says it has to be strictly less than 255.